src/elisp/treemacs-extensions.el

```diff
@@ -447,28 +449,33 @@ additional keys."
                          ,(format "The handle to update the the %s extension." name))
                        (defun ,ext-name ()
                          (treemacs-with-writable-buffer
-                          (-let [pr (make-treemacs-project
+                          (let ((pr (make-treemacs-project
                                      :name ,root-label
                                      :path ,root-key-form
-                                     :path-status 'extension)]
+                                     :path-status 'extension))
+                                (button-start (point)))
                             (treemacs--set-project-position ,root-key-form (point-marker))
                             (setq-local ,project-var-name pr)
-                            (insert (propertize "a"
+                            (insert (propertize "\n"
                                                 'button '(t)
                                                 'category 'default-button
+                                                'invisible t
+                                                'cursor-sensor-functions (list (lambda (_ previous-position change)
+                                                                                 (when (eq change 'entered)
+                                                                                   (if (< previous-position (point))
+                                                                                       (treemacs-next-line 1)
+                                                                                     (treemacs-previous-line 1))
+                                                                                   (hl-line-highlight))))
                                                 :custom t
                                                 :key ,root-key-form
                                                 :path (list :custom ,root-key-form)
                                                 :depth -1
                                                 :project pr
                                                 :state ,closed-state-name))
-                            (goto-char 0)
-                            (funcall ',expand-name)
-                            (goto-char 0)
-                            ;; (narrow-to-region (1+ (point-at-eol)) (point-max))
-                            (setq-local hide-ov (make-overlay (point-at-bol) (1+ (point-at-eol))))
-                            (overlay-put hide-ov 'invisible t)
-                            ))))
+                            (save-excursion
```

Comment:
Not sure what's the correct behavior here. Probably not this... I believe the point should actually end up after the expanded nodes.